fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -506,6 +531,70 @@ private StoreTimer.Counter getCounter(@Nonnull final FDBRecordContext recordCont
         return Verify.verifyNotNull(Verify.verifyNotNull(recordContext.getTimer()).getCounter(event));
     }
 
+    @Test
+    void basicPartitionTest() {
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+            recordStore.saveRecord(createComplexDocument(6666L, ENGINEER_JOKE, "kilroy was here", 2, 2, false, Instant.now().toEpochMilli()));
+            recordStore.saveRecord(createComplexDocument(7777L, WAYLON, "ereh saw yorlik", 1, 2, true, Instant.now().plus(1, ChronoUnit.DAYS).toEpochMilli()));
+            assertIndexEntryPrimaryKeyTuples(Set.of(Tuple.from(2, 6666L)),
+                    recordStore.scanIndex(COMPLEX_PARTITIONED, fullTextSearch(COMPLEX_PARTITIONED, "\"propose a Vision\""), null, ScanProperties.FORWARD_SCAN));
+            assertEquals(1, getCounter(context, FDBStoreTimer.Counts.LOAD_SCAN_ENTRY).getCount());
+
+            Subspace partition0Subspace = recordStore.indexSubspace(COMPLEX_PARTITIONED).subspace(TupleHelpers.EMPTY.add(LucenePartitioner.PARTITION_DATA_SUBSPACE).add(0));
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition0Subspace, context, "_0.cfs");
+        }
+    }
+
+    private static void joinedPartitionedLuceneIndexMetadataHook(@Nonnull final RecordMetaDataBuilder metaDataBuilder) {
+
+        //set up the joined index
+        final JoinedRecordTypeBuilder joined = metaDataBuilder.addJoinedRecordType("luceneJoinedPartitionedIdx");
+        joined.addConstituent("complex", "ComplexDocument");
+        joined.addConstituent("simple", "SimpleDocument");
+        joined.addJoin("simple", field("group"), "complex", field("group"));
+
+        metaDataBuilder.addIndex(joined, new Index("joinNestedConcat", concat(
+                field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
+                field("simple").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text"))
+                )
+        ), LuceneIndexTypes.LUCENE, ImmutableMap.of(IndexOptions.TEXT_DOCUMENT_PARTITION_TIMESTAMP, "complex.time")));
+        metaDataBuilder.addIndex("SimpleDocument", "simple$group", field("group"));
+    }
+
+    @Test
+    void partitionedJoinedIndexTest() {
```

Comment:
It would be good to also have a test that has multiple groups, and is partitioned.